File: code/ex161.java
class Solution {
    public boolean isOneEditDistance(String s, String t) {
        int ns = s.length(), nt = t.length();
        if (Math.abs(ns - nt) > 1)    return false;
        int cnt = 0, ps = 0, pt = 0;
        while (ps < ns && pt < nt) {
            if (s.charAt(ps) != t.charAt(pt)) {
                cnt++;
                if (ns == nt) {ps++;    pt++;}
                else if (ns < nt)    pt++;
                else    ps++;
            } else {
                ps++;    pt++;
            }
            if (cnt > 1)    return false;
        }
        cnt += ns - ps + nt - pt;
        return cnt == 1;
    }
}

File: code/ex161.cpp
class Solution {
public:
    bool isOneEditDistance(string s, string t) {
        int ns = s.length(), nt = t.length();
        if (ns > nt) {
            string temp = s;
            s = t;
            t = temp;
        }
        
        bool changed = 0;
        int is, it;
        for (is = 0, it = 0; is < ns || it < nt; ) {
            if (s[is] == t[it]) {
                it++;
                is++;
            } else {
                if (ns == nt)    is++;
                it++;
                if (changed)    return 0;
                changed = 1;
            }
        }
        
        return changed;
    }
};